src/monitoring/request_timer.rs

```diff
@@ -0,0 +1,30 @@
+use rocket::fairing::{Fairing, Info, Kind};
+use chrono::Utc;
+use rocket::{Request, Response, Data};
+use rocket::http::Header;
+
+pub struct RequestTimer();
+
+impl Fairing for RequestTimer {
+    fn info(&self) -> Info {
+        Info {
+            name: "RequestTimer",
+            kind: Kind::Request | Kind::Response,
+        }
+    }
+
+    fn on_request(&self, request: &mut Request, _data: &Data) {
+        let instant = Utc::now().timestamp_millis();
+        log::debug!("on_request: {}", &instant);
+        let h = Header::new("requested_timestamp", instant.to_string());
```

Comment:
Hmmm do we have a better alternative than the header? (it is fine if everything else is way more complicated)

Some notes:
- I would make the header key a constant
- Maybe look at `local_cache` from the `Request` struct